XenModel/Utils/Helpers.cs

```diff
@@ -959,31 +959,120 @@ public static void RemoveFromGuiConfig(Session session, IXenObject o, string key
         static Regex PifEthRegex = new Regex("^pif_eth([0-9]+)_(tx|rx)((_errors)?)$");
-		static Regex PifVlanRegex = new Regex("^pif_eth([0-9]+).([0-9]+)_(tx|rx)((_errors)?)$");
+        static Regex PifVlanRegex = new Regex("^pif_eth([0-9]+).([0-9]+)_(tx|rx)((_errors)?)$");
         static Regex PifBrRegex = new Regex("^pif_xenbr([0-9]+)_(tx|rx)((_errors)?)$");
-		static Regex PifXapiRegex = new Regex("^pif_xapi([0-9]+)_(tx|rx)((_errors)?)$");
+        static Regex PifXapiRegex = new Regex("^pif_xapi([0-9]+)_(tx|rx)((_errors)?)$");
         static Regex PifTapRegex = new Regex("^pif_tap([0-9]+)_(tx|rx)((_errors)?)$");
         static Regex PifLoRegex = new Regex("^pif_lo_(tx|rx)((_errors)?)$");
         static Regex PifBondRegex = new Regex("^pif_(bond[0-9]+)_(tx|rx)((_errors)?)$");
+        static Regex PifOtherRegex = new Regex("^pif_aggr_(tx|rx)$");
         static Regex DiskRegex = new Regex("^vbd_((xvd|hd)[a-z]+)(_(read|write))?(_latency)?$");
         static Regex DiskIopsRegex = new Regex("^vbd_((xvd|hd)[a-z]+)_iops_(read|write|total)$");
         static Regex DiskThroughputRegex = new Regex("^vbd_((xvd|hd)[a-z]+)_io_throughput_(read|write|total)$");
         static Regex DiskOtherRegex = new Regex("^vbd_((xvd|hd)[a-z]+)_(avgqu_sz|inflight|iowait)$");
         static Regex NetworkLatencyRegex = new Regex("^network/latency$");
         static Regex XapiLatencyRegex = new Regex("^xapi_healthcheck/latency$");
+        static Regex XapiMemoryRegex = new Regex("^xapi_(allocation|free_memory|live_memory|memory_usage)_kib$");
         static Regex StatefileLatencyRegex = new Regex("^statefile/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}/latency$");
         static Regex LoadAvgRegex = new Regex("loadavg");
-    	static Regex SrRegex = new Regex("^sr_[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}_cache_(size|hits|misses)");
+        static Regex SrRegex = new Regex("^sr_[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}_cache_(size|hits|misses)");
         static Regex SrIORegex = new Regex("^(io_throughput|iops)_(read|write|total)_([a-f0-9]{8})$");
         static Regex SrOtherRegex = new Regex("^(latency|avgqu_sz|inflight|iowait)_([a-f0-9]{8})$");
         static Regex SrReadWriteRegex = new Regex("^((read|write)(_latency)?)_([a-f0-9]{8})$");
         static Regex GpuRegex = new Regex(@"^gpu_((memory_(free|used))|power_usage|temperature|(utilisation_(compute|memory_io)))_((([a-fA-F0-9]{4}\/)|([a-fA-F0-9]{8}\/))?[a-fA-F0-9]{2}\/[0-1][a-fA-F0-9].[0-7])$");
 
+        public static DataSourceCategory GetDataSourceCategory(string name)
+        {
+            if (CpuRegex.IsMatch(name) || CpuAvgFreqRegex.IsMatch(name) ||
+                CpuStateRegex.IsMatch(name) || CpuOtherRegex.IsMatch(name))
+                return DataSourceCategory.Cpu;
+
+            if (VcpuRegex.IsMatch(name))
+                return DataSourceCategory.Cpu;

```

Comment:
These can be rewritten as a single if statement.

Same applies to line 1067 with

```csharp
if (XapiMemoryRegex.IsMatch(name))
                return DataSourceCategory.Memory;

if (name.StartsWith("memory"))
                return DataSourceCategory.Memory;
```